components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImplIT.java

```diff
@@ -1028,5 +1028,149 @@ public void testUpdateComment() throws Exception {
+            apiDAO.addAPI(api);
+        }
+
+        // Filter APIs by single status
+        List<String> singleStatus = new ArrayList<>();
+        singleStatus.add(publishedStatus);
+
+        List<API> apiList = apiDAO.getAPIsByStatus(singleRole, singleStatus);
+
+        Assert.assertTrue(APIUtils.isListsEqualIgnoreOrder(apiList, publishedAPIsSummaryAdmin, new APIComparator()));
+
+        List<String> twoStatus = new ArrayList<>();
+
+        twoStatus.add(createdStatus);
+        twoStatus.add(blockedStatus);
+        List<String> twoRoles = new ArrayList<>();
+        twoRoles.add(ADMIN);
+        twoRoles.add("creator");
+        // Add APIs
+        List<API> publishedAPIsSummaryTwoRoles = new ArrayList<>();
+        testAddGetEndpoint();
+        for (int i = 0; i < numberOfPublishedCreator; ++i) {
+            API api = SampleTestObjectCreator.createUniqueAPI().lifeCycleStatus(publishedStatus).
+                    visibleRoles(twoRoles).build();
+            publishedAPIsSummaryTwoRoles.add(SampleTestObjectCreator.getSummaryFromAPI(api));
+            apiDAO.addAPI(api);
+        }
+
+        List<API> createdAPIsSummaryTwoRoles = new ArrayList<>();
+        for (int i = 0; i < numberOfCreatedCreator; ++i) {
+            API api = SampleTestObjectCreator.createUniqueAPI().lifeCycleStatus(createdStatus).
+                    visibleRoles(twoRoles).build();
+            createdAPIsSummaryTwoRoles.add(SampleTestObjectCreator.getSummaryFromAPI(api));
+            apiDAO.addAPI(api);
+        }
+
+        List<API> blockedAPIsSummaryTwoRoles = new ArrayList<>();
+        for (int i = 0; i < numberOfBlockedCreator; ++i) {
+            API api = SampleTestObjectCreator.createUniqueAPI().lifeCycleStatus(blockedStatus).
+                    visibleRoles(twoRoles).build();
+            blockedAPIsSummaryTwoRoles.add(SampleTestObjectCreator.getSummaryFromAPI(api));
+            apiDAO.addAPI(api);
+        }
+        apiList = apiDAO.getAPIsByStatus(twoRoles, twoRoles);
+
+        Assert.assertEquals(apiList.size(), publishedAPIsSummaryTwoRoles.size() + blockedAPIsSummaryTwoRoles.size());
+
+        for (API api : publishedAPIsSummaryTwoRoles) {
+            Assert.assertTrue(apiList.contains(api));
+            apiList.remove(api);
+        }
+
+        for (API api : blockedAPIsSummaryTwoRoles) {
+            Assert.assertTrue(apiList.contains(api));
+            apiList.remove(api);
+        }
+
+        Assert.assertTrue(apiList.isEmpty());
+
+    }
+
+    @Test(expectedExceptions = APIMgtDAOException.class)
+    public void testGetAPIByStatusException() throws APIMgtDAOException {
+
+        List<String> role = new ArrayList<>();
+        List<String> statuses = new ArrayList<>();
+
+        ApiDAO apiDAO = DAOFactory.getApiDAO();
+        apiDAO.getAPIsByStatus(role, statuses);
+
+        List<API> apiList = apiDAO.getAPIsByStatus(role, statuses);
+
```

Comment:
Remove unwanted blank lines. Check the whole PR. There are multiple places to do so.